testing/tests/DevExpress.data/applyChanges.tests.js

```diff
@@ -0,0 +1,110 @@
+import { extend } from 'core/utils/extend';
+import applyChanges from 'data/apply_changes';
+
+QUnit.module('Apply Changes', {
```

Comment:
Why no test for the case when keyExpr is not defined and no 'id' field in objects?

Reply:
I'll implement it as a separate task. We need to discuss how we will notify a user. Moreover, applyBatch is used by our data stores and they do not throw errors in the console un such cases.